packages/wallet/src/wallet.ts

```diff
@@ -304,6 +327,57 @@ export class DemexWallet {
     messages: EncodeObject[],
     signOpts?: SignTxOpts,
     broadcastOpts?: BroadcastTxOpts
+  ): Promise<BroadcastTxResult> {
+    await this.initWalletAccountState(this.signer);
+    const checkAccMergedStatus = !!this.triggerMerge || !!signOpts?.triggerMerge;
+    if (checkAccMergedStatus) await this.triggerWalletAccMergeIfRequired(messages);
+    return await this.queueTx(messages, signOpts, broadcastOpts);
+  }
+
+
+  private async triggerWalletAccMergeIfRequired(messages: EncodeObject[]) {
+    const walletAddress = this.bech32Address;
+    const hexPublicKey = this.publicKey.toString("hex");
+    const mergeOwnAccountMessage = this.containsMergeWalletAccountMessage(messages);
+    if (mergeOwnAccountMessage) return;
+    const isAccountMerged = this.walletAccounts[walletAddress]?.isMerged;
+    if (isAccountMerged) return
+    const message: EncodeObject = {
+      typeUrl: TxTypes.MsgMergeAccount,
+      value: MsgMergeAccount.fromPartial({
+        // get address via mapping as account may exist only as evm bech32
+        creator: this.walletAccounts[walletAddress]?.address,
+        pubKey: hexPublicKey,
+      }),
+    }
+    await this.queueTx([message]);
+    await this.reloadWalletAccountPostSuccessMerge();
+  }
+
+  private containsMergeWalletAccountMessage(messages: readonly EncodeObject[]) {
+    const mergeAccountMessage = findMessageByTypeUrl(messages, TxTypes.MsgMergeAccount);
+    if (mergeAccountMessage) {
+      const { value } = mergeAccountMessage;
+      const msg = value as MsgMergeAccount;
+      const mergeOwnAccountMessage =
+        (msg.creator === this.bech32Address
+          || msg.creator === this.evmBech32Address)
+        && this.publicKey.toString("hex") === msg.pubKey;
+      if (mergeOwnAccountMessage) return true;
+    }
+    return false;
+  }
+
+  private async reloadWalletAccountPostSuccessMerge() {
+    await this.reloadAccountFromSigner(this.signer);
+    const address = await getDefaultSignerAddress(this.signer);
+    this.updateWalletAccount(address, { isMerged: true });
+  }
```

Comment:
`this.updateWalletAccount` is async, and it also calls `this.reloadAccountFromSigner`. isnt that a double execution?

Reply:
function name also has too much context, would prefer a more neutral name that describes the work done, instead of who calls it.

Reply:
this.updateWalletAccount is not async? hmm

Reply:
refactored the function name